src/allocation_engine/interval_tree.rs

```diff
@@ -441,11 +452,18 @@ impl InnerNode {
         let range_start = match constraint.policy {
             AllocPolicy::FirstMatch => align_up(node_key.start(), constraint.align)?,
             AllocPolicy::LastMatch => {
+                // This operation can not underflow as we check at the beginning
```

Comment:
Hmm, the second part of this paragraph is a bit hard to read ... is it basically saying that adding `1` towards the end of the chain cannot overflow because we previously subtracted a `size > 0` from the same value?